libgalois/include/katana/PropertyFileGraph.h

```diff
@@ -240,6 +276,8 @@ class KATANA_EXPORT PropertyFileGraph {
   }
 
   /// Determine if two PropertyFileGraphss are Equal
+  // NB: It might be useful to have a version where property tables
+  // with permuted columns compare as equal
```

Comment:
I did not catch the drift of this. What kind of permutations?

Also, what's NB?

Reply:
nota bene: Latin for "please note"

Reply:
Though, most of the time, I find you can basically drop "NB" and keep the same intent without adding a dependency on Latin :p

Reply:
Haha, yes :)

Reply:
But the NB enabled this delightful discussion.

My point about permutations is this.  Let's say I have two node properties, A and B.  In the `node_properties` table, column A comes first.  If I update an A property, writing the table requires removing the column, updating it, and adding it back to the table.  Now, column B comes first.  Even if I change the values in A to be what they were originally, Arrow won't compare the two node property tables as equal.

I propose listing all the columns by name and checking them individually for equality.  I think that is a better notion of `Equal` than what we have now (arrow::Array::Equal).  I know I would use it.  Maybe @arthurp would.

Reply:
I understand now. Yes, I agree with your proposal. That makes more sense.

Reply:
I have added this functionality to this PR.